libs/core/src/lib/checkbox/checkbox/checkbox.component.ts

```diff
@@ -0,0 +1,124 @@
+import {Component, ElementRef, EventEmitter, forwardRef, Input, Output, Renderer2} from '@angular/core';
+import {CheckboxControlValueAccessor, ControlValueAccessor, NG_VALUE_ACCESSOR} from '@angular/forms';
+import {FdCheckboxValues} from './fd-checkbox-values.interface';
+
+let checkboxUniqueId: number = 0;
+
+@Component({
+    selector: 'fd-checkbox',
+    templateUrl: './checkbox.component.html',
+    styleUrls: ['./checkbox.component.scss'],
+    providers: [
+        {
+            provide: NG_VALUE_ACCESSOR,
+            useExisting: forwardRef(() => CheckboxComponent),
+            multi: true
+        }
+    ]
+})
+export class CheckboxComponent implements ControlValueAccessor {
+
+    /** Sets id property of input, binds input with input label */
+    @Input()
+    inputId: string = `fd-checkbox-${checkboxUniqueId++}`;
+
+    /** State of input, changes visual appearance of input.
+     * Available states: 'valid', 'invalid', 'info', 'warning'
+     * */
+    @Input()
+    state: 'valid' | 'invalid' | 'info' | 'warning';
+
+    /** Sets name property of input */
+    @Input()
+    name: string;
+
+    /** Sets content of input label */
+    @Input()
+    label: string;
+
+    /** Allows to disable/enable input */
+    @Input()
+    disabled: boolean;
+
+    /** Allows to minimize input to compact mode */
+    @Input()
+    compact: boolean;
+
+    /** Enables third state of the input */
+    @Input()
+    tristate: boolean;
+
+    /** Allows to prevent user from manually selecting third state of the input */
+    @Input()
+    tristateSelectable: boolean = true;
+
+    /** Values returned by checkbox.
+     * By default checkbox returns true(checked), false(unchecked), null(third state) values.*/
+    @Input()
+    values: FdCheckboxValues = {trueValue: true, falseValue: false, thirdStateValue: null};
+
+    /** Emits new checkbox value whenever its changed */
+    @Output()
+    onChange: EventEmitter<any> = new EventEmitter<any>();
+
+    public checkboxValue: any;
+    public checkboxState: 'checked' | 'unchecked' | 'indeterminate';
+    public onTouched = () => {};
+    public onValueChange = (newValue) => {};
+
+    get isChecked(): boolean {
+        return this.checkboxState === 'checked';
+    }
+
+    get isIndeterminate(): boolean {
+        return this.checkboxState === 'indeterminate';
+    }
+
+    writeValue(value: any): void {
```

Comment:
Let's mark methods, that are implemented from `ControlValueAccessor` as `/** @hidden */`, they shouldn't be in documentation IMO. Let's shortly describe all of other methods.

Reply:
Done